alternator/executor.cc

```diff
@@ -577,16 +579,16 @@ future<> verify_permission(
                     schema->ks_name(), schema->cf_name(), username));
             }
         });
+    });
 }
 
 // Similar to verify_permission() above, but just for CREATE operations.
 // Those do not operate on any specific table, so require permissions on
 // ALL KEYSPACES instead of any specific table.
 future<> verify_create_permission(const service::client_state& client_state) {
-    return client_state.check_has_permission(auth::command_desc(
-            auth::permission::CREATE,
-            auth::resource(auth::resource_kind::data))).then(
-        [&client_state] (bool allowed) {
+    return do_with(auth::resource(auth::resource_kind::data), [&client_state] (const auto& resource) {
```

Comment:
would be better to fix by corutiniazation than do_with

Reply:
Interesting idea. I deliberately didn't use a coroutine (it's a new function, not some old function from the day of continuions) - because I wanted to avoid an allocation. But... if I already have an allocation for this stupid "resource" thing, might as well just make it a coroutine!

Reply:
Done.